### DISCUSSION TITLE
NSPANEL after flashing on esphome and connecting to 230V, the Sonoff firmware is running again

### DISCUSSION THREAD

**Discussion by siminkuv** [Blackymas/NSPanel_HA_Blueprint#2425] [General]
When I flashed from esphome to homeassistant, the ESPHOME firmware was loaded. Until I flashed, the display was even on 3.3V and the USB converter. Since the flash, the display has not been on, but only flashing via Wifi works, not via USB flash. I tried connecting the board back to the main board with access to 230V, but the display started with the Sonoff logo and their interface. Does the board have two firmware? Thank you

**Comment by edwardtfn** [comment]
> Does the board have two firmware?

I don't think this is the case, bit it does. The bootloader keeps two copies of the firmware (an old one and the latest) for the case the OTA fails.
It should manage this and only boot from the latest, but it can (unlikely) happen some issue that drives the boot to the incorrect partition.
Flashing it again should solve this issue.

But if yiu are understanding the Sonoff firmware is running due to what you see in the screen, that mi might not be correct.
Before opening your panel again, please try to find it under Settings > Devices and Services > ESPHome. If you find it, just select the model and click "Update TFT" as that will update the firmware in the display itself (it has its own controller separated from ESP32) and that should do the trick.

**Comment by alphabravocharly72** [comment]
hi,
flashed the NSPanel. Can turn on/off the two relays via the integration. But if I push the button for Update TFT (Model Selected) nothing happend. What I do wrong?
After a few minutes the display restarts a few times but always with the "old" Display in few.
I flashed a few NSPanels all with the same issue.

![image](https://github.com/user-attachments/assets/2c635fe0-1716-4c8c-ac1f-424eeae977d3)

![image](https://github.com/user-attachments/assets/0b8a034e-e51b-4d73-a488-c39af8da2884)

![image](https://github.com/user-attachments/assets/1e802a4a-a3ef-43b5-8f8b-f33608590378)

![image](https://github.com/user-attachments/assets/c0c72aa2-868c-4517-9919-4558be153241)

Thx for helping

**Reply by edwardtfn** [reply]
> But if I push the button for Update TFT (Model Selected) nothing happend. What I do wrong?

We need your logs to be able to help here.

On the ESPHome Dashboard, click on the "Logs" button related to your device, and keep that logging window open, then with another tab, click on "Update TFT". Wait a couple of minutes, then collect the logs and share it here, please.

**Comment by alphabravocharly72** [comment]
Here is the log.

INFO ESPHome 2024.12.2
INFO Reading configuration /config/esphome/wall-display-134cec.yaml...
INFO Updating https://github.com/Blackymas/NSPanel_HA_Blueprint.git@main
INFO Updating https://github.com/Blackymas/NSPanel_HA_Blueprint@v4.3.12
INFO Detected timezone 'Europe/Vienna'
INFO Starting log output from 192.168.178.145 using esphome API
INFO Successfully connected to wall-display-134cec @ 192.168.178.145 in 0.042s
WARNING wall-display-134cec @ 192.168.178.145: Connection error occurred: wall-display-134cec @ 192.168.178.145: The connection dropped immediately after encrypted hello; Try enabling encryption on the device or turning off encryption on the client (ESPHome Logs 2024.12.2).
WARNING Can't connect to ESPHome API for wall-display-134cec @ 192.168.178.145: wall-display-134cec @ 192.168.178.145: The connection dropped immediately after encrypted hello; Try enabling encryption on the device or turning off encryption on the client (ESPHome Logs 2024.12.2). (HandshakeAPIError)
INFO Trying to connect to wall-display-134cec @ 192.168.178.145 in the background
INFO Successfully connected to wall-display-134cec @ 192.168.178.145 in 0.073s
WARNING wall-display-134cec @ 192.168.178.145: Connection error occurred: wall-display-134cec @ 192.168.178.145: The connection dropped immediately after encrypted hello; Try enabling encryption on the device or turning off encryption on the client (ESPHome Logs 2024.12.2).
INFO Successfully connected to wall-display-134cec @ 192.168.178.145 in 0.074s
WARNING wall-display-134cec @ 192.168.178.145: Connection error occurred: wall-display-134cec @ 192.168.178.145: The connection dropped immediately after encrypted hello; Try enabling encryption on the device or turning off encryption on the client (ESPHome Logs 2024.12.2).
INFO Successfully connected to wall-display-134cec @ 192.168.178.145 in 0.022s
WARNING wall-display-134cec @ 192.168.178.145: Connection error occurred: wall-display-134cec @ 192.168.178.145: The connection dropped immediately after encrypted hello; Try enabling encryption on the device or turning off encryption on the client (ESPHome Logs 2024.12.2).
INFO Successfully connected to wall-display-134cec @ 192.168.178.145 in 0.078s
WARNING wall-display-134cec @ 192.168.178.145: Connection error occurred: wall-display-134cec @ 192.168.178.145: The connection dropped immediately after encrypted hello; Try enabling encryption on the device or turning off encryption on the client (ESPHome Logs 2024.12.2).
INFO Successfully connected to wall-display-134cec @ 192.168.178.145 in 0.015s
WARNING wall-display-134cec @ 192.168.178.145: Connection error occurred: wall-display-134cec @ 192.168.178.145: The connection dropped immediately after encrypted hello; Try enabling encryption on the device or turning off encryption on the client (ESPHome Logs 2024.12.2).
INFO Successfully connected to wall-display-134cec @ 192.168.178.145 in 0.120s
WARNING wall-display-134cec @ 192.168.178.145: Connection error occurred: wall-display-134cec @ 192.168.178.145: The connection dropped immediately after encrypted hello; Try enabling encryption on the device or turning off encryption on the client (ESPHome Logs 2024.12.2).

**Comment by alphabravocharly72** [comment]
Hi,
Is this the correct log?
Thx

**Comment by alphabravocharly72** [comment]
Any news?

**Comment by cthiele-mogic** [comment]
I have the same issue with an new panel, I followed the instructions (many tries) - even bought a new Raspi 5 with 8 GB to compile the firmware. Everything without problems, firmware installed via serial interface.  Back on 230V it shows again the Sonoff screen/firmware...

**Reply by cthiele-mogic** [reply]
Ok, "Update TFT" helped after restarting

**Comment by alphabravocharly72** [comment]
tried again to fix it.
No success.
Can any one help me please
Thx

Looked to find the tft file for upload. But cant find it. Where can I find the file?

**Reply by chrismaddahl** [reply]
https://github.com/Blackymas/NSPanel_HA_Blueprint/tree/main/hmi

**Reply by chrismaddahl** [reply]
I had the same problem, after initial flashing the sonof screen was still showing.
So I open the log file in esphome and clicked in another tab the "Update TFT Display" button.
And the upload started.
![2025-03-19 19_50_39-Testboard – Home Assistant – Mozilla Firefox](https://github.com/user-attachments/assets/34c586e0-4ea1-437d-a763-f420897c981a)
![20250319_195033_cr](https://github.com/user-attachments/assets/997f3eb5-8f58-4f4d-a1d4-2fb1a6dccac8)

and now it is working perfectly.

**Comment by Neicke29** [comment]
Hi, after flashing two display, both have the same problem. connecting with power, both displays show the sonoff screen. I can switch the hardswitches or changing brightness via home assistant, but pressing "update TFT display" nothing happend. I flashed now several times. I also check, that nspanel_eu.tft is in the www folder before flashing. (wasn`t there before the first few tries). But nothing changed. Displays are always in Sonoff Display mode.
Can anyone help me

![Screenshot 2026-02-10 153440](https://github.com/user-attachments/assets/fed8062f-f41c-48d9-96a4-f11b1d154b69)

**Reply by edwardtfn** [reply]
Could you please collect the logs from the moment you press the "Upload TFT" button until the time your panel restarts (or even a bit after that)?
Please copy the full log text and paste here, instead of a screenshot (which is quite limited).

**Reply by edwardtfn** [reply]
This `Invalid variable name` is not important now. It is the consequence of using a different TFT, so after replacing the TFT that should disappear. We can ignore it for now.

**Reply by Neicke29** [reply]
Hi,
thanks for reply. As i now tried to do the lock and pressed update tft, tft is now starting again and again new, but suddenly it shows an upload screen. So just waiting now what to happen next.
[logs_nspanel1_logs (1).txt](https://github.com/user-attachments/files/25253613/logs_nspanel1_logs.1.txt)
Maybe now its fixed.

**Reply by Neicke29** [reply]
Ok, HA is working now on the display as it seems. But when i am now in the NSPanel configuration in settings->blueprints and i change anything, it doesn't save on the panel when i press save. What i am doing wrong?
Thanks for your support so far.

**Reply by edwardtfn** [reply]
Do you see anything related to this in your Home Assistant logs (Settings > System > Logs)?
Do you mind creating an issue here in the repo specifically for this and provide more details there?

**Reply by Neicke29** [reply]
Hi, so sorry for my impatient so far. I was not expecting that updates to the panel also need a while. So far, everything ist working as expected. Great software for this panel and exact that, what i was looking for.
